src/mbed_os_tools/detect/darwin.py

```diff
@@ -40,7 +42,12 @@ def _plist_from_popen(popen):
     if not out:
         return []
     try:
-        return loads(out)
+        # Beautiful soup ensures the XML is properly formed
+        soup = BeautifulSoup(out.decode('utf8'), 'xml')
+        if soup.get_text() == '':
```

Comment:
Sometimes it is safer in Python to check for falseness rather than explicit values.
```suggestion
        if not soup.get_text():
```